bundles/org.openhab.binding.airq/src/main/java/org/openhab/binding/airq/internal/AirqHandler.java

```diff
@@ -0,0 +1,815 @@
+                thStatus = ThingStatus.OFFLINE;
+            } else {
+                logger.warn("air-Q - airqHandler - getData(): retried but still cannot reach the air-Q device.");
+            }
+        } else {
+            if (thStatus == ThingStatus.OFFLINE) {
+                updateStatus(ThingStatus.ONLINE);
+                thStatus = ThingStatus.ONLINE;
+            }
+        }
+        return res;
+    }
+
+    // does the networking job (and only that)
+    protected @Nullable Result doNetwork(String address, String requestMethod, @Nullable String body) {
+        int timeout = 10000;
+        HttpURLConnection conn = null;
+        logger.debug("air-Q - airqHandler - doNetwork(): connecting to {} with method {} and body {}", address,
+                requestMethod, body);
+        try {
+            conn = (HttpURLConnection) new URL(address).openConnection();
+            conn.setRequestMethod(requestMethod);
+            conn.setRequestProperty("Content-Type", "application/json");
+            conn.setConnectTimeout(timeout);
+            conn.setReadTimeout(timeout);
+            if (body != null && !body.isEmpty()) {
+                conn.setDoOutput(true);
+                try (Writer out = new OutputStreamWriter(conn.getOutputStream())) {
+                    out.write(body);
+                }
+            }
+            try (InputStream in = conn.getInputStream(); ByteArrayOutputStream result = new ByteArrayOutputStream()) {
+                byte[] buffer = new byte[1024];
+                int length;
+                while ((length = in.read(buffer)) != -1) {
+                    result.write(buffer, 0, length);
+                }
+                conn.disconnect();
+                return new Result(result.toString(StandardCharsets.UTF_8.name()), conn.getResponseCode());
+            }
+        } catch (IOException exc) {
+            logger.warn("air-Q - airqHandler - doNetwork(): Error while accessing air-Q: {}", exc.toString());
+        } finally {
+            if (conn != null) {
+                conn.disconnect();
+            }
+        }
+        return null;
+    }
+
+    public static class Result {
+        private final String body;
+        private final int responseCode;
+
+        public Result(String body, int responseCode) {
+            this.body = body;
+            this.responseCode = responseCode;
+        }
+
+        public String getBody() {
+            return body;
+        }
+
+        public int getResponseCode() {
+            return responseCode;
+        }
+    }
+
+    @Override
+    public void dispose() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+        }
+        if (getConfigDataJob != null) {
+            getConfigDataJob.cancel(true);
+        }
+    }
+
+    public void pollData() {
+        logger.trace("air-Q - airqHandler - run(): starting polled data handler");
+        if ((!config.ipAddress.isEmpty()) && (!config.password.isEmpty())) {
```

Comment:
config.ipAddress and config.password can not be empty as you have marked them as required in the XML file. The handler will not start until the framework is happy so you have this check multiple times and you can remove it which will free up CPU time. Use the find/search feature to find all cases.